shard-operator-parent/shard-operator-core/src/test/java/com/redhat/service/smartevents/shard/operator/core/TemplateProvider.java

```diff
@@ -0,0 +1,25 @@
+package com.redhat.service.smartevents.shard.operator.core;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.utils.Serialization;
+
+public class TemplateProvider {
```

Comment:
If you make [this method](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/providers/TemplateProviderImpl.java#L136) protected you should be able to extend the existing `TemplateProvider` class to limit the duplication. Wdyt?

In any case, please rename the class to avoid to clash with the other `TemplateProvider` class